src/audio/play.rs

```diff
@@ -2,19 +2,20 @@
 
 use std::io::Cursor;
 
-use rodio::Decoder;
-use rodio::play_once as rplay_once;
+use rodio::{Decoder, play_once as rplay_once, Sink, Source as RSource};
 
 use super::DecoderError;
 use super::output::Output;
 use super::source::Source;
 
-/// Play a sound once.
+/// Play a sound once.  A volume of 1.0 is unchanged, while 0.0 is silent.
 ///
 /// This will return an Error if the loaded audio file in source could not be decoded.
-pub fn try_play_once(source: &Source, endpoint: &Output) -> Result<(), DecoderError> {
-    match rplay_once(&endpoint.endpoint, Cursor::new(source.clone())) {
-        Ok(sink) => {
+pub fn try_play_once(source: &Source, volume: f32, endpoint: &Output) -> Result<(), DecoderError> {
+    let mut sink = Sink::new(&endpoint.endpoint);
+    match Decoder::new(Cursor::new(source.clone())) {
+        Ok(source) => {
+            sink.append(source.amplify(volume));
```

Comment:
Is there any difference between `sink.set_volume` and `source.amplify`?

Reply:
Not really no, they both have the same impact on emitted samples.  So I'm just going to leave this as is because there's no reason to go either direction.

Reply:
So I guess there is one subtle difference.  Sink volumes can take up to 5 ms to be propagated to the sink, so source amplify doesn't have this 5 ms delay.